src/pipelinedefinition.cpp

```diff
@@ -784,6 +790,43 @@ Status PipelineDefinition::validateForCycles() {
     return StatusCode::OK;
 }
 
+Status PipelineDefinition::validateDemultiplexerGatherNodesOrder() {
+    auto it = std::find_if(std::begin(nodeInfos), std::end(nodeInfos), [](const NodeInfo& nodeInfo) { return nodeInfo.kind == NodeKind::EXIT; });
+    std::vector<std::pair<std::string, std::vector<std::set<std::string>>>> nodesToCheck{{it->nodeName, {}}};
+    if (!it->gatherFromNode.empty()) {
+        nodesToCheck.back().second.emplace_back(it->gatherFromNode);
+    }
+    while (!nodesToCheck.empty()) {
+        auto [node, demultiplyStack] = nodesToCheck.back();
```

Comment:
Rename node to nodeName so it is clear that this is name (string) identifier and not actual NodeInfo instance